tests/test_samples.py

```diff
@@ -1339,3 +1339,38 @@ def test_old_gui():
         make_2d_axes(['x0', 'y0'], tex={'x0': '$x_0$', 'y0': '$y_0$'})
     with pytest.raises(NotImplementedError):
         make_1d_axes(['x0', 'y0'], tex={'x0': '$x_0$', 'y0': '$y_0$'})
+
+
+def test_groupby_stats():
+    mcmc = read_chains('./tests/example_data/cb')
+    chains = mcmc.groupby('chain')
+    for chain in [1, 2]:
+        i = mcmc.chain == chain
+        assert_allclose(mcmc.loc[i].mean().drop('chain'),
+                        chains.mean().loc[chain, :])
+        assert_allclose(mcmc.loc[i].std().drop('chain'),
+                        chains.std().loc[chain, :])
+        assert_allclose(mcmc.loc[i].median().drop('chain'),
+                        chains.median().loc[chain, :])
+        assert_allclose(mcmc.loc[i].var().drop('chain'),
+                        chains.var().loc[chain, :])
+
+    assert_allclose(mcmc.mean().drop('chain'), chains.mean().mean())
```

Comment:
Looks like this works, cool! :)

Reply:
Can we get the same to work with `var`? (Might not be possible...)

Reply:
Nope, I don't think that makes actually sense...

Reply:
I *think* that what @williamjameshandley has done with setting weights in all of mean, var etc means that mean of means, variance of variances will be correctly calculated taking weights into consideration. 

I think you're right that the mean is somewhat special in that the mean of means is the same as the overall mean, but there might be similar tests we can construct for the others

Reply:
The only one this works for (as far as I can see) is the mean of means. The mean of variances/variance of variances fundamentally fails because they correctly use the cluster means to compute their separate variances. There may be use-cases for this distinction in @Ormorod's clustering work.